### ISSUE TITLE
Slight refactor, eventually

### ISSUE THREAD

**Issue by patrick-kidger** [google-research/torchsde#22]
Not suggesting that this happen now, so I'm just opening an issue to keep track of it, but I think the current hierarchy of:
```
_brownian
_core
    methods
brownian_lib
_brownian_lib [from C++]
```
(not counting `_core/adjoint_sdes` which I know you've removed in your latest branch)

should be switched to:
```
_impl
    brownian
    methods
    core
```
(So additionally tidying the Brownian things together somehow. If we end up implementing Brownian-everything in C++ then that would suffice, for example.)

**Comment by lxuechen** [comment]
Agree. This refactoring seems sensible, and should come in when we have the C++ implementations.
